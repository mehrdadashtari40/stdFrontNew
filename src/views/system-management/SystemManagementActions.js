import getAuthonticatedJSON from '../../common/utils/functions/getAuthenticatedJSON';

export const FETCH_CATEGORIES = 'FETCH_CATEGORIES';
export const FETCH_CATEGORY = 'FETCH_CATEGORY';
export const LOAD_USER = 'LOAD_USER';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_DEPARTMENTS = 'LOAD_DEPARTMENTS';
export const LOAD_DEPARTMENT = 'LOAD_DEPARTMENT';



export function loadCategories(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategories(response));
		});
	};
}

export function setCategories(data) {
	return {
		type: FETCH_CATEGORIES,
		Categories: data,
	};
}

export function loadCategory(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategory(response));
		});
	};
}

export function setCategory(data) {
	return {
		type: FETCH_CATEGORY,
		Category: data,
	};
}

export function loadUser(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setUser(response));
		});
	};
}

export function setUser(data) {
	return {
		type: LOAD_USER,
		User: data,
	};
}

export function loadGroups(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setGroups(response));
		});
	};
}

export function setGroups(data) {
	return {
		type: LOAD_GROUPS,
		Groups: data,
	};
}

export function loadDepartments(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartments(response));
		});
	};
}

export function setDepartments(data) {
	return {
		type: LOAD_DEPARTMENTS,
		Departments: data,
	};
}


export function loadDepartment(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartment(response));
		});
	};
}

export function setDepartment(data) {
	return {
		type: LOAD_DEPARTMENT,
		Department: data,
	};
}


