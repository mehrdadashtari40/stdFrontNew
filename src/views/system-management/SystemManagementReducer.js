import {
	FETCH_CATEGORIES,
	FETCH_CATEGORY,
	LOAD_USER,
	LOAD_GROUPS,
	LOAD_DEPARTMENTS,
	LOAD_DEPARTMENT,
} from './SystemManagementActions';

const initialState = {
	Categories: [],
	Category: [],
	User: [],
	Users: [],
	Groups: [],
	Departments: [],
	Department: [],
};

export function systemManagementReducer(state = initialState, action) {
	switch (action.type) {
		case FETCH_CATEGORIES:
			return {
				...state,
				Categories: action.Categories,
			};

		case FETCH_CATEGORY:
			return {
				...state,
				Category: action.Category,
			};

		case LOAD_USER:
			return {
				...state,
				User: action.User,
			};
		case LOAD_GROUPS:
			return {
				...state,
				Groups: action.Groups,
			};
		case LOAD_DEPARTMENTS:
			return {
				...state,
				Departments: action.Departments,
			};
		case LOAD_DEPARTMENT:
			return {
				...state,
				Department: action.Department,
			};

		default:
			return state;
	}
}
