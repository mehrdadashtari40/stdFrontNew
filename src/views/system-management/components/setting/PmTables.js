import React from 'react'
import {config} from '../../../../config/config';
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import MaterialDataTable from "../../../../common/tables/components/MaterialDataTable";
import Form from "../../utils/CategoryForm";

export default class PmTables extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: false,
        };
    }

    render() {
        const {loading} = this.state;
        let IframeURL = config.iframeServer + "pmTables" + "?sid=" + localStorage.getItem('session_id');

        return (<>
                <Card id="content">
                    <CardContent>
                        <div className="row">
                            <div className="col-sm-12 col-md-12 col-lg-12 p-0">
                                <iframe name="my-iframe" src={IframeURL} id="my-iframe"
                                        style={{width: "100%", height: "1000px", marginTop: "0", border: "none"}}
                                        sandbox="allow-downloads allow-top-navigation allow-same-origin allow-scripts allow-forms allow-modals allow-popups"></iframe>
                            </div>
                        </div>
                    </CardContent>
                </Card>
            </>
        )
    }

}