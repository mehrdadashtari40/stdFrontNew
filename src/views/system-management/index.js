import Categories from './components/setting/Categories';
import PmTables from './components/setting/PmTables';
import Backup from './components/backup/Backup';
import Setting from './components/setting/Setting';
import Users from './components/users/Users';
import Groups from './components/users/Groups';
import Department from './components/users/Departments';
export const routes = [
	{
		path: '/process-categories',
		exact: true,
		component: Categories,
		name: 'Categories',
	},
	{
		path: '/pm-tables',
		exact: true,
		component: PmTables,
		name: 'PmTables',
	},
	{
		path: '/backup',
		exact: true,
		component: Backup,
		name: 'Backup',
	},
	{
		path: '/setting',
		exact: true,
		component: Setting,
		name: 'Setting',
	},
	{
		path: '/users',
		exact: true,
		component: Users,
		name: 'Users',
	},
	{
		path: '/groups',
		exact: true,
		component: Groups,
		name: 'Groups',
  },
  
  {
		path: '/departments',
		exact: true,
		component: Department,
		name: 'Department',
	},
];
export * from './SystemManagementActions';
export * from './SystemManagementReducer';
