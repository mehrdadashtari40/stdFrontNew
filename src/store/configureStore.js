import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
