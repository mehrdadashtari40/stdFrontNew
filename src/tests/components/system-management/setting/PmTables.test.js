import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/configureStore';
import configureStore from 'redux-mock-store';
import * as selectActions from '../../../../views/system-management/SystemManagementActions';
import PmTables from '../../../../views/system-management/components/setting/PmTables';

// const store = mockStore();

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , PmTables Component', () => {
	const wrapper = shallow(<PmTables/>);
	expect(wrapper).toMatchSnapshot();
});
