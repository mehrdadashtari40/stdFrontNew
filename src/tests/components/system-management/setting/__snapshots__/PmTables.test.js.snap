// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , PmTables Component 1`] = `
<div
  className="animated fadeInDown"
  id="main"
  role="main"
>
  <iframe
    id="my-iframe"
    name="my-iframe"
    sandbox="allow-top-navigation allow-same-origin allow-scripts allow-forms allow-modals allow-popups"
    src="http://ls.arian.co.ir:8081/syszums/fa/modern/pmTables?sid=null"
    style={
      Object {
        "border": "none",
        "height": "1000px",
        "marginTop": "80px",
        "width": "100%",
      }
    }
  />
</div>
`;
